### PR TITLE
quiche: implement QuicSystemEventLoopImpl

### PR DESCRIPTION
**Pull request by danzh2010**
Add a dummy impl to QuicSystemEventLoop which is used to drive SimpleEpollServer to receive IO event in Chromium, but not needed in stand alone e2e tests.

Risk Level: low
Testing: added a test to make sure it compiles
Part of #2557

### CODE REVIEW

File: test/extensions/quic_listeners/quiche/platform/quic_system_event_loop_impl.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+inline void QuicRunSystemEventLoopIterationImpl() {
+  // No-op.
+}
+
+class QuicSystemEventLoopImpl {
+public:
+  // Only used by quic_client_bin.cc which is not required in Envoy.
```
Comment (alyssawilk) [inline test/extensions/quic_listeners/quiche/platform/quic_system_event_loop_impl.h]: lazy question - if it's only used by a file not required by Envoy, why can't we just skip implementing it?

Comment (danzh2010) [reply test/extensions/quic_listeners/quiche/platform/quic_system_event_loop_impl.h]: QuicRunSystemEventLoopIterationImpl is used by QuicClientEpollNetworkHelper which is needed to run quic e2e tests. Though QuicSystemEventLoopImpl is not used, it is bundled in same API file as QuicRunSystemEventLoopIterationImpl, so it need to be defined.

Comment (alyssawilk) [reply test/extensions/quic_listeners/quiche/platform/quic_system_event_loop_impl.h]: Ah, that's a bummer - thanks for the explanation.  

LGTM modulo clang tidy warning, and master merge.

Comment (danzh2010) [reply test/extensions/quic_listeners/quiche/platform/quic_system_event_loop_impl.h]: done

**Pull request merged**